server/planning/agendas_async/agendas_async_service.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any
+from planning.core.service import BasePlanningAsyncService
+from planning.types import AgendasResourceModel
+from planning.common import set_original_creator
+from apps.auth import get_user_id
+from superdesk import get_resource_service
+from superdesk.errors import SuperdeskApiError
+from superdesk.notification import push_notification
+
+
+class AgendasAsyncService(BasePlanningAsyncService[AgendasResourceModel]):
+    # TODO-ASYNC: on_fetched mechanism to be added to the base REST API class
+    # async def _generate_planning_info(self, docs):
+    #     # TODO-ASYNC: Change this when async planning service is done
+    #     planning_service = get_resource_service("planning")
+    #     for doc in docs:
+    #         doc["plannings"] = planning_service.get_planning_by_agenda_id(doc.get(ID_FIELD)).docs
+    #
+    # async def on_fetched(self, docs):
+    #     await self._generate_planning_info(docs.get(ITEMS))
+    #
+    # async def on_fetched_item(self, doc):
+    #     await self._generate_planning_info([doc])
+
+    async def on_create(self, docs: list[AgendasResourceModel]) -> None:
+        for doc in docs:
+            doc_dict = doc.to_dict()
+            set_original_creator(doc_dict)
```

Comment:
I'm not sure this is going to work. Once you convert the pydantic model instance into a dictionary, there are two different objects, the changes made to the dict won't be reflected in the model instance, therefore the data won't be saved into the database. Or maybe I'm missing something?

Reply:
This should be added in the base Planning service, as we do in [Newshub](https://github.com/superdesk/newsroom-core/blob/async/newsroom/core/resources/service.py#L22), for both create and update hooks